src/rpc/request.rs

```diff
@@ -695,6 +703,25 @@ pub enum FundingInfo {
     Monero(MoneroFundingInfo),
 }
 
+#[derive(Clone, PartialEq, Eq, Debug, Display, StrictEncode, StrictDecode)]
+#[display("{swap_id}, {public_offer}")]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+#[display(NodeInfo::to_yaml_string)]
+pub struct CheckpointEntry {
+    pub swap_id: SwapId,
+    pub public_offer: PublicOffer<BtcXmr>,
+}
```

Comment:
I didn't quite get why pub offer is in this struct?

Reply:
It's so users can see for which taken offers a swap was running. I might remove this again at a later stage, once we decide to either keep with the manual method, or automate everything and don't surface manual restoration to the user.

Reply:
in farcasterd/Runtime there is consumed_offers field that tracks what offers were consumed

https://github.com/farcaster-project/farcaster-node/blob/4d6a27b5124be9755f8b02071ea2cfd65607040f/src/farcasterd/runtime.rs#L170